### ISSUE TITLE
Improve `TESTING.md` doc

### ISSUE THREAD

**Issue by AbihaFatima** [publiclab/plots2#10928]
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new** and looking through our general [help wanted](https://github.com/publiclab/plots2/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Claim the issue by commenting. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!
<details>
  <summary>💡 <i>Learn how to claim</i> 🙋</summary>
  <h3>Claiming an issue</h3>
  <p>Unless the issue is marked as reserved for someone, you can just say "I'd like to try this!" and then you've claimed it - no need to wait for someone to assign it to you. Just be sure you link your pull request (PR) to this issue so we can see where your solution is.</p>
  <p>And open one early if possible - even before you've completed it with additional commits - and others can help you figure out any issues you may face.</p>
</details>

- [ ] 📝 **Update** the file [TESTING.md](https://github.com/publiclab/plots2/blob/main/doc/TESTING.md) in the `plots2` repository (press the little pen Icon) and edit the line as shown below.

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -1,6 +1,6 @@
 (Moved from https://github.com/publiclab/plots2/wiki/Testing)

-Run all basic rails tests with `rails test -d`. This is **no longer required for submitting pull requests** (see [Pull Requests](#pull-requests), below), and to confirm you have a working local environment.
+Run all basic rails tests with `rails test -d`. This is **no longer required for submitting pull requests** (see [Pull Requests](#pull-requests) below) and to confirm you have a working local environment.

 `rake test:all` runs teaspoon-mocha client-side tests and coverage reporting.

@@ -138,7 +138,7 @@ test:

 6 - Run rake `db:setup`

-If everything run smoothly, this will avoid some weird errors (like passing the
+If everything runs smoothly, this will avoid some weird errors (like passing the
 tests locally but not on travis). Remember *not to add* those to your commits.

 ## Tests with MySQL features
@@ -179,7 +179,7 @@ But, don't worry, feel free to ping @publiclab/maintainers on your open PR to re

 Whether or not you've tested your code on the unstable server, once your PR is merged into `plots2/main`, it auto-triggers the build of the [stable version](https://stable.publiclab.org) of the website. This is the "pre-production" server and the last chance to test out any changes before they're published to the live site.

-While it builds, on either the unstable or stable server, you'll be redirected to see the progress of the build in our [Jenkins](https://jenkins.laboratoriopublico.org/) console. Once it's complete, you can try accessing the original URL once more to try it out. See Build Status, below.
+While it builds, on either the unstable or stable server, you'll be redirected to see the progress of the build in our [Jenkins](https://jenkins.laboratoriopublico.org/) console. Once it's complete, you can try accessing the original URL once more to try it out. See [Build Status](#build-status) below.
 This is the last chance for you to run any manual tests that you have, and the last chance to find any bugs. Do checkout the stable build once your PR is acccepted!

@@ -193,4 +193,4 @@ The exact workflow is this:<br>
 After some time (usually a week or two), one of the maintainers deploys the code to the [live site](https://publiclab.org). Usually, they send a message in the [chatroom](http://publiclab.org/chat) to notify other contributors of the update. Your code is now live!

 ## Build Status
-You can monitor all the buils at [Jenkins](https://jenkins.laboratoriopublico.org/)
+You can monitor all the builds at [Jenkins](https://jenkins.laboratoriopublico.org/)
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://app.egghead.io/playlists/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there.

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance!

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it.

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below, and [reach out in our chatroom](https://publiclab.org/chat)!

**Comment by AbihaFatima** [comment]
Hey @TildaDares! here's a FTO issue, kindly have a look. Thanks!

**Comment by TildaDares** [comment]
@AbihaFatima Great work spotting this. There’s also an error in the Build Status heading, can you add that to the issue too? https://github.com/publiclab/plots2/blob/main/doc/TESTING.md#build-status. `Builds` is spelt as `buils` there. Thanks!!

**Comment by AbihaFatima** [comment]
> @AbihaFatima Great work spotting this. There’s also an error in the Build Status heading, can you add that to the issue too? https://github.com/publiclab/plots2/blob/main/doc/TESTING.md#build-status. `Builds` is spelt as `buils` there. Thanks!!

Done! 👍

**Comment by TildaDares** [comment]
Thanks @AbihaFatima!!

**Comment by TildaDares** [comment]
Reserved for @stylebytemi for 24 hours!!

**Comment by YUVRAJ06singh08deora** [comment]
I would like to try this issue please assign me this issue.
